lab-tim/lib/storage.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const debug = require('debug')('http:storage');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+
+const storage = module.exports = {};
+
+// fs.readFileProm(`${__dirname}/../data/${schema}/${item_id}.json)
+// .then(...)
+// .catch(...)
+
+// const memory = {
+//   'toy': {
+//     '123-456-789': {_id: '123-456-789', name: 'barney', desc: 'purple dino'}
+//   }
+// }
+
+storage.create = function(schema, item) {
+  debug('#storage.create');
+  // debugger
+  return new Promise(function(resolve, reject) {
+    if(!schema) return reject(new Error('cannot create; schema required'));
+    if(!item) return reject(new Error('cannot create; item required'));
```

Comment:
Because we are creating a toy before validating that something exists, there will always be an item here, even if you send incomplete data. To fix this, and probably solve a few of the test bugs, you would want to move the creation of a new toy into this function.